### PR TITLE
Docs: Update reference configurations YAML templates for Struct deprecation for Any and hosts deprecation for load_assignment

### PR DESCRIPTION
**Pull request by moderation**
Description: Update examples for Struct deprecation for Any
Risk Level: Low - generated configs only, no changes to code
Testing: `bazel build //configs:example_configs`, `bazel test //test/...`
Docs Changes: None required
Release Notes: None required
Completes: https://github.com/envoyproxy/envoy/pull/6025
Replaces: https://github.com/envoyproxy/envoy/pull/6356
Related: https://github.com/envoyproxy/envoy/pull/6346

### CODE REVIEW

**Comment by moderation** [comment]
Ready for review @dio

File: configs/envoy_double_proxy_v2.template.yaml [hunk 1]
```diff
@@ -42,14 +43,18 @@
...
         - name: envoy.health_check
-          config:
+          typed_config:
+            '@type': type.googleapis.com/envoy.config.filter.http.health_check.v2.HealthCheck
             pass_through_mode: false
-            endpoint: /healthcheck
-          name: envoy.buffer
-          config:
+            headers:
+              - exact_match: /healthcheck
```
Comment (htuch) [inline configs/envoy_double_proxy_v2.template.yaml]: How come this changed?

File: configs/envoy_double_proxy_v2.template.yaml [hunk 2]
```diff
@@ -91,20 +97,30 @@ static_resources:
...
   - name: backhaul
     type: STRICT_DNS
     connect_timeout: 1s
     lb_policy: ROUND_ROBIN
-    hosts:
-    - socket_address:
-        protocol: TCP
-        address: front-proxy.yourcompany.net
-        port_value: 9400
+    load_assignment:
```
Comment (htuch) [inline configs/envoy_double_proxy_v2.template.yaml]: So, this is a bit more than just switching to `Any`, can you update the commit message and PR title to cover all the things that are changing?

File: configs/envoy_double_proxy_v2.template.yaml (hunk 1 above)
Comment (moderation) [reply configs/envoy_double_proxy_v2.template.yaml]: The healthcheck configuration has changed. It used to take a `endpoint: /healthcheck` but when you switch to the `Any` format healthcheck it requires the `headers` format - see https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/health_check/v2/health_check.proto#envoy-api-msg-config-filter-http-health-check-v2-healthcheck

File: configs/envoy_double_proxy_v2.template.yaml (hunk 2 above)
Comment (moderation) [reply configs/envoy_double_proxy_v2.template.yaml]: It's true that I'm also changing the deprecated `hosts:` to `load_assignment:`. Will update title.

**Comment by moderation** [comment]
@htuch The resulting YAML is tested with `bazel test //test/...` and it has been a painstaking process to run all the changes through to eliminate all the issues when working through the changes. Further than that I did run all of the generated config files in `generated/configs/` to make sure they loaded into Envoy. This combined with passing tests should be OK.

**Comment by moderation** [comment]
The documentation doesn't change in this PR but the referring section in the docs to the Reference configurations is at https://www.envoyproxy.io/docs/envoy/latest/install/ref_configs

File: configs/envoy_double_proxy_v2.template.yaml (hunk 2 above)
Comment (dio) [reply configs/envoy_double_proxy_v2.template.yaml]: Thanks for this though, so when I resurrect this: https://github.com/envoyproxy/envoy/issues/4618 I don't have to deal with the `host` to `load_assignment` updates for examples.

File: configs/envoy_front_proxy_v2.template.yaml [hunk 3]
```diff
@@ -31,7 +31,8 @@
   {%endif %}
     filters:
     - name: envoy.http_connection_manager
-      config:
+      typed_config:
+        '@type': type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
```
Comment (dio) [inline configs/envoy_front_proxy_v2.template.yaml]: Super nit, not sure if we should always use `"` instead of `'` for this.

Comment (moderation) [reply configs/envoy_front_proxy_v2.template.yaml]: Good catch. Fixing now.

**Pull request merged**